### PR TITLE
Swin faster rcnn

### PR DESCRIPTION
**Pull request by Chen-Song**
- add swin transformer backbone
- add swin-t faster rcnn 1x  (mAP 42.7)
- add swin-t faster rcnn 2x  (mAP 44.8)
- add swin-t faster rcnn 3x (mAP 45.3)

- please see https://github.com/PaddlePaddle/PaddleDetection/pull/4050

### CODE REVIEW

File: ppdet/modeling/backbones/__init__.py [hunk 1]
```diff
@@ -18,24 +18,23 @@
...
+from . import swin_transformer_tiny
+from .swin_transformer_tiny import *
 
 from .vgg import *
 from .resnet import *
 from .darknet import *
 from .mobilenet_v1 import *
 from .mobilenet_v3 import *
 from .hrnet import *
-from .lite_hrnet import *
```
Comment (lyuwenyu) [inline ppdet/modeling/backbones/__init__.py]: Do not modify none-related code

File: configs/faster_rcnn/_base_/faster_rcnn_swin_transformer_tiny.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  embed_dim: 96
+  depths: [2, 2, 6, 2]
+  num_heads: [3, 6, 12, 24]
+  window_size: 7
+  ape: false
+  drop_path_rate: 0.1
+  patch_norm: true
+  out_indices: [0,1,2,3]
+  drop_path_rate: 0.1
+  pretrained: https://bj.bcebos.com/v1/ai-studio-online/19a72dd9eb884f4581492a61fab901e60e858e34569f4805b619eceabd6a4315?responseContentDisposition=attachment%3B%20filename%3Dswin_tiny_patch4_window7_224.pdparams
```
Comment (qingqing01) [inline configs/faster_rcnn/_base_/faster_rcnn_swin_transformer_tiny.yml]: 这个链接太长了, @lyuwenyu 可以和其他backone存一起吧

File: ppdet/modeling/backbones/swin_transformer_tiny.py [hunk 3]
```diff
@@ -0,0 +1,728 @@
+import numpy as np
```
Comment (qingqing01) [inline ppdet/modeling/backbones/swin_transformer_tiny.py]: add copyright

File: ppdet/modeling/backbones/swin_transformer_tiny.py [hunk 4]
```diff
@@ -0,0 +1,728 @@
...
+            zeros_(m.bias)
+            ones_(m.weight)
+
+    def forward(self, x):
+        """Forward function."""
+        # dict_keys(['im_id', 'is_crowd', 'gt_bbox', 'curr_iter', 'image', 'im_shape', 'scale_factor', 'target0', 'target1', 'target2'])
+        x = self.patch_embed(x['image'])
+        # Wh, Ww = x.size(2), x.size(3)
+        _, _, Wh, Ww = x.shape
+        if self.ape:  # 不会到这一步
```
Comment (qingqing01) [inline ppdet/modeling/backbones/swin_transformer_tiny.py]: change to EN comments

File: configs/faster_rcnn/_base_/faster_rcnn_swin_transformer_tiny.yml (hunk 2 above)
Comment (lyuwenyu) [reply configs/faster_rcnn/_base_/faster_rcnn_swin_transformer_tiny.yml:20]: > 这个链接太长了, @lyuwenyu 可以和其他backone存一起吧

好的

Comment (lyuwenyu) [reply configs/faster_rcnn/_base_/faster_rcnn_swin_transformer_tiny.yml]: done

File: ppdet/modeling/backbones/swin_transformer_tiny.py (hunk 4 above)
Comment (lyuwenyu) [reply ppdet/modeling/backbones/swin_transformer_tiny.py]: done

File: ppdet/modeling/backbones/swin_transformer_tiny.py (hunk 3 above)
Comment (lyuwenyu) [reply ppdet/modeling/backbones/swin_transformer_tiny.py]: done

**Pull request closed without merging**